libs/shared/flagd-core/src/lib/flagd-core.ts

```diff
@@ -91,35 +92,47 @@ export class FlagdCore {
       }
     }
 
-    const defaultVariant = flag.defaultVariant;
-
-    let resolvedVariant;
+    let variant;
     let reason;
 
-    if (flag.targetingString == undefined) {
-      resolvedVariant = flag.variants.get(defaultVariant);
+    if (!flag.targeting) {
+      variant = flag.defaultVariant;
       reason = StandardResolutionReasons.STATIC;
     } else {
-      // todo - targeting evaluation
-      // till targeting is handled, return the static result
-      resolvedVariant = flag.variants.get(defaultVariant);
-      reason = StandardResolutionReasons.STATIC;
+      let targetingResolution;
+      try {
+        targetingResolution = this._targeting.applyTargeting(flagKey, flag.targeting, evalCtx);
+      } catch (e) {
+        console.error(`Error evaluating targeting rule for flag ${flagKey}, falling back to default`, e);
+        targetingResolution = null;
+      }
+
+      if (!targetingResolution) {
+        variant = flag.defaultVariant;
+        reason = StandardResolutionReasons.DEFAULT;
+      } else {
+        variant = targetingResolution;
+        reason = StandardResolutionReasons.TARGETING_MATCH;
+      }
+    }
+
+    if (typeof variant !== 'string') {
```

Comment:
flagd casts the variant to a string. That allows for the shorthand JSON logic implementation like `{"==" : [1, 1]}` to match the `true` variant.

Reply:
We should make sure this is explicitly defined in the spec and tested since it's a common pattern in flagd.

FYI @toddbaert

Reply:
Do you mean the variant key? Anything but a string would be invalid JSON, and if you mean the `defaultVariant`, the schema doesn't allow it to be anything but a string

Reply:
I understand now, this is the targeting result. I think this is worth specifying. Could you open an issue in flagd?

Reply:
I added special toString() conversion for targeting resolving. Plus added a test to validate the short circuiting rule